plugins/source/vault/client/spec.go

```diff
@@ -0,0 +1,30 @@
+package client
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+)
+
+const defaultConcurrency = 10000
+
+type Spec struct {
+	VaultAddress string `json:"vault_address,omitempty"`
+	Concurrency  int    `json:"concurrency"`
+}
+
+func (s *Spec) SetDefaults() {
+	if s.Concurrency == 0 {
```

Comment:
```suggestion
	if s.Concurrency <= 0 {
```

(nit) in absence of validation that the value is non-negative